src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -186,6 +217,28 @@ private async Task ProcessQueueAsync()
                     var context = await work.CreateRequestContextAsync(cancellationToken).ConfigureAwait(false);
                     if (work.MutatesServerState)
                     {
+                        if (CancelInProgressWorkUponMutatingRequest)
+                        {
+                            if (pendingTasks.Count > 0)
```

Comment:
can remove this check.

Reply:
I had the check there to prevent the ToArray and Task.WhenAll calls, but it probably isn't worth it. Removing.